library/src/engine.cxx

```diff
@@ -143,32 +143,62 @@ void engine::loadPlugin(const std::string& pathOrName)
     }
     else
     {
-      std::string libName;
+      std::vector<std::string> searchPaths;
+
+      std::string envValue;
+      if (vtksys::SystemTools::GetEnv("F3D_PLUGINS_PATH", envValue))
+      {
+        // split path with OS separator (':' on Linux/macOS and ';' on Windows)
+#ifdef _WIN32
+        char sep = ';';
+#else
+        char sep = ':';
+#endif
+        vtksys::SystemTools::Split(envValue, searchPaths, sep);
+      }
 
-      // Right now, plugins should be located in the same folder than f3d.exe on Windows
-      // On Linux/macOS, when we are not using a full path, we rely on LD_LIBRARY_PATH
+      // On Windows, search plugins in the same folder than f3d.exe too by default
 #ifdef _WIN32
       std::vector<wchar_t> pathBuf(40000);
       if (GetModuleFileNameW(
             GetModuleHandle(nullptr), pathBuf.data(), static_cast<DWORD>(pathBuf.size())))
       {
         std::string progPath =
           vtksys::SystemTools::GetProgramPath(vtksys::Encoding::ToNarrow(pathBuf.data()));
-        libName = vtksys::SystemTools::ConvertToWindowsOutputPath(progPath) + "\\";
+        searchPaths.push_back(progPath);
       }
 #endif
 
       // construct the library file name from the plugin name
-      libName += vtksys::DynamicLoader::LibPrefix();
+      std::string libName = vtksys::DynamicLoader::LibPrefix();
       libName += "f3d-plugin-";
       libName += pathOrName;
       libName += vtksys::DynamicLoader::LibExtension();
 
-      handle = vtksys::DynamicLoader::OpenLibrary(libName);
+      // try search paths
+      for (const std::string& path : searchPaths)
+      {
+        std::string tryPath = path + '/' + libName;
+        tryPath = vtksys::SystemTools::ConvertToOutputPath(tryPath);
+
+        handle = vtksys::DynamicLoader::OpenLibrary(tryPath);
+
+        if (handle)
+        {
+          // plugin is found and loaded
+          break;
```

Comment:
could we add an example test relying on this mechanism ?

Reply:
I don't see any easy way to cover this.